test/functional/wallet_encryption.py

```diff
@@ -47,6 +49,25 @@ def run_test(self):
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrase, '', 1)
         assert_raises_rpc_error(-8, "passphrase cannot be empty", self.nodes[0].walletpassphrasechange, '', 'ff')
 
+        # Make sure there are no unencrypted key records
+        # Use the wallet tool's dump mechanism to get all of the records to inspect them.
+        self.nodes[0].unloadwallet(self.default_wallet_name)
+        dump_file = os.path.join(self.nodes[0].datadir_path, "wallet.dump")
+        subprocess.check_call([self.options.bitcoinwallet, f"-datadir={self.nodes[0].datadir_path}", f"-chain={self.chain}", f"-wallet={self.default_wallet_name}", f"-dumpfile={dump_file}", "dump"], stdout=subprocess.DEVNULL)
+        self.nodes[0].loadwallet(self.default_wallet_name)
+        # Look for records that contain the hex for 'key' but not 'ckey'
+        key_hex = b"key".hex()
+        ckey_hex = b"ckey".hex()
```

Comment:
in e3c292b298e5e6f2ec319fb93ba67fd64a0f4ac1

Relying that all encrypted keys contain `ckey` seems fragile. Can we maybe list all the encrypted key records?

Reply:
I think it's more futureproof to check for `ckey` since all new records involving encrypted keys have always had that suffix.